### PR TITLE
Use the AsyncLogger from the Logentries library

### PR DESCRIPTION
**Pull request by vgeshel**
Instead of doing our own socket management.

### CODE REVIEW

**Comment by vgeshel** [comment]
This seems to be working well

File: deps.edn [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{:aliases
+ {:1.7  {:extra-deps #:org.clojure {clojure #:mvn {:version "1.7.0"}}},
+  :1.8  {:extra-deps #:org.clojure {clojure #:mvn {:version "1.8.0"}}},
+  :1.9  {:extra-deps #:org.clojure {clojure #:mvn {:version "1.9.0"}}},
+  :1.10 {:extra-deps #:org.clojure {clojure #:mvn {:version "1.20.0"}}}
```
Comment (tanzoniteblack) [inline deps.edn]: `1.20.0`?

File: deps.edn [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  :runner
+  {:extra-deps
+   #:com.cognitect {test-runner
+                    {:git/url
+                     "https://github.com/cognitect-labs/test-runner",
+                     :sha "76568540e7f40268ad2b646110f237a60295fa3c"}},
+   :main-opts      ["-m" "cognitect.test-runner" "-d" "test"]},
+  :test {:extra-deps  {org.clojure/test.check #:mvn {:version "RELEASE"}}
+         :extra-paths ["test"]}
+  :dev  {:jvm-opts ["-Dclojure.spec.skip-macros=true"
```
Comment (tanzoniteblack) [inline deps.edn]: Why are we skipping spec checks of the macros?

File: .travis.yml [hunk 3]
```diff
@@ -1,4 +1,6 @@
-language: clojure
-script: lein test-all
+language: generic
+services:
+  - docker
+script: sh -c 'docker run -i --rm -v $(pwd):/work -w /work clojure:openjdk-8-tools-deps-alpine clojure -A:test:runner'
```
Comment (tanzoniteblack) [inline .travis.yml:4]: Not that we have great tests in this project, but the previous version ran tests against all supported versions of clojure (`1.{7,8,9,10}`)

**Review by tanzoniteblack — CHANGES_REQUESTED** [review]

File: deps.edn (hunk 1 above)
Comment (vgeshel) [reply deps.edn]: fixed

File: deps.edn (hunk 2 above)
Comment (vgeshel) [reply deps.edn]: fixed

File: .travis.yml (hunk 3 above)
Comment (vgeshel) [reply .travis.yml:4]: @tanzoniteblack do you have an example for doing this with `deps.edn`?

Comment (tanzoniteblack) [reply .travis.yml:4]: I do not. A quick google search doesn't find anything, so I think we'll have to write a bash loop or something similar?

**Comment by vgeshel** [comment]
The travis builds passed. I think there is a glitch on their side and the status didn't propagate to github.

File: src/yummly/logentries_timbre_appender.clj [hunk 4]
```diff
@@ -66,46 +55,54 @@
...
-  the format `<TOKEN> MESSAGE>`, like datadog."
+  This uses com.logentries.net.AsyncLogger, which is the underlying implementation of the log4j and logback appenders from Logentries. That class uses a bounded queue and may drop messages under heavy load. When that happens, it will write error messages to stderr, but only if `:debug?` is `true`. See https://github.com/rapid7/le_java/blob/master/src/main/java/com/logentries/net/AsyncLogger.java.
+
+  If an exception happens during logging, this appender will catch and not rethrow, meeting the standard expectation of a logging library. Exceptions will be logged to stderr at a rate of no more of 1 per minutes per appender. Additional information on the frequency of exxceptions may be found my inspecting the appender (see the fields `:call-count` and `:error-count`.
+
+  Note that `cheshire.core` is used to serialize log messages to json. If something in your `:user-tags` or `:context` is not readily serializable by `cheshire`, this will cause exceptions and those messages *will not* be logged. See https://github.com/dakrone/cheshire#custom-encoders for how to teach `chechire` to encode your custom data."
```
Comment (tanzoniteblack) [inline src/yummly/logentries_timbre_appender.clj:71]: small typo: "for how to teach `chechire`" --> "for how to teach `cheshire`"

**Review by tanzoniteblack — APPROVED** [review]

**Review by tanzoniteblack — APPROVED** [review]

**Pull request merged**